src/pages/Home.js

```diff
@@ -0,0 +1,352 @@
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import Header from "../components/Header";
+import Footer from "../components/Footer";
+import ClampLines from "react-clamp-lines";
+import BannerCarousel from "../components/BannerCarousel";
+import { useData } from "../context/DataProvider";
+import ThumbnailView from "../components/ThumbnailView";
+
+export default function Home() {
+  const {
+    state: { banners },
+    dispatch: dataDispatch,
+  } = useData();
+
+  const [trendingBooks, setTrendingBooks] = useState([]);
+  const [bestsellerBooks, setBestsellerBooks] = useState([]);
+  const [newBooks, setNewBooks] = useState([]);
+
+  const getNewBooks = async () => {
```

Comment:
These functions can be put in utils,making components compact as possible.Something like ```utils/pages/Home``` can have all the functions defined on this component and can be imported from there.